go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go

```diff
@@ -0,0 +1,227 @@
+
+import (
+	"io"
+	"sort"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/diff"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/env/actions"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/index"
+)
+
+// UnscopedDiffTable is a sql.Table implementation of a system table that shows which tables have
+// changed in each commit, across all branches.
+type UnscopedDiffTable struct {
+	ddb  *doltdb.DoltDB
+	head *doltdb.Commit
+}
+
+// NewUnscopedDiffTable creates an UnscopedDiffTable
+func NewUnscopedDiffTable(_ *sql.Context, ddb *doltdb.DoltDB, head *doltdb.Commit) sql.Table {
+	return &UnscopedDiffTable{ddb: ddb, head: head}
+}
+
+// Name is a sql.Table interface function which returns the name of the table which is defined by the constant
+// LogTableName
+func (dt *UnscopedDiffTable) Name() string {
+	return doltdb.DiffTableName
+}
+
+// String is a sql.Table interface function which returns the name of the table which is defined by the constant
+// DiffTableName
+func (dt *UnscopedDiffTable) String() string {
+	return doltdb.DiffTableName
+}
+
+// Schema is a sql.Table interface function that returns the sql.Schema for this system table.
+func (dt *UnscopedDiffTable) Schema() sql.Schema {
+	return []*sql.Column{
+		{Name: "commit_hash", Type: sql.Text, Source: doltdb.LogTableName, PrimaryKey: true},
+		{Name: "committer", Type: sql.Text, Source: doltdb.LogTableName, PrimaryKey: false},
+		{Name: "email", Type: sql.Text, Source: doltdb.LogTableName, PrimaryKey: false},
+		{Name: "date", Type: sql.Datetime, Source: doltdb.LogTableName, PrimaryKey: false},
+		{Name: "message", Type: sql.Text, Source: doltdb.LogTableName, PrimaryKey: false},
+		{Name: "table_name", Type: sql.Text, Source: doltdb.LogTableName, PrimaryKey: false},
+	}
+}
+
+// Partitions is a sql.Table interface function that returns a partition of the data. Currently data is unpartitioned.
+func (dt *UnscopedDiffTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
+	return index.SinglePartitionIterFromNomsMap(nil), nil
+}
+
+// PartitionRows is a sql.Table interface function that gets a row iterator for a partition.
+func (dt *UnscopedDiffTable) PartitionRows(ctx *sql.Context, _ sql.Partition) (sql.RowIter, error) {
+	return NewUnscopedDiffTableItr(ctx, dt.ddb, dt.head)
+}
+
+// UnscopedDiffTableItr is a sql.RowItr implementation which iterates over each commit as if it's a row in the table.
+type UnscopedDiffTableItr struct {
+	ctx          *sql.Context
+	ddb          *doltdb.DoltDB
+	commits      []*doltdb.Commit
+	commitIdx    int
+	tableNames   []string
+	tableNameIdx int
+}
+
+// NewUnscopedDiffTableItr creates a UnscopedDiffTableItr from the current environment.
+func NewUnscopedDiffTableItr(ctx *sql.Context, ddb *doltdb.DoltDB, head *doltdb.Commit) (*UnscopedDiffTableItr, error) {
+	commits, err := actions.TimeSortedCommits(ctx, ddb, head, -1)
```

Comment:
Fine for a first pass, but you probably want an iterator that fetches commits one at a time, rather than this which gets them all at once

Reply:
Sounds good; I've got the two perf improvements noted in my queue and will follow up with a separate PR for them: 1)don't load all commits up front, and 2) implemented sql.FilteredTable interface.